### ISSUE TITLE
Incorrect JSON data format passed to prompts.py file

### ISSUE THREAD

**Issue by maciejbrasewicz** [promptfoo/promptfoo#173]
I encountered an issue with the data being passed to the `prompts.py` script. The expectation was that the data would be in a proper JSON format, but the received data showed formatting errors.

**Steps to Reproduce:**

**1. Creating prompt_generator file:**

```
%%writefile prompt_generator.py

import sys
import json

def generate_prompt(context):
    document = context["vars"]["document"]
    question = context["vars"]["question"]
    
    return f"""Document.
{document}
Question: {question}
"RELEVANT" or "NOT_RELEVANT":"""

if __name__ == '__main__':
    data = sys.argv[1] if len(sys.argv) > 1 else None
    with open("log_data.txt", "a") as f:
        f.write(f"Log data: {data}\n")
    print(generate_prompt(json.loads(data)))
```

**2. Creating prompfooconfig.yaml file:**

```
%%writefile promptfooconfig.yaml
prompts:
  - 'prompt_generator.py:generate_prompt'
providers: 
  - openai:gpt-3.5-turbo
tests:
  - vars:
      question: "question1"
      document: "document1"
    assert:
      - type: equals
        value: 'NOT_RELEVANT'
  - vars:
      question: "question2"
      document: "document2"
    assert:
      - type: equals
        value: 'RELEVANT'
```
        
**Log data file:**       

```
Log data: '{vars:{question:question1,document:document1}}'
Log data: '{vars:{question:question2,document:document2}}'
```

It should be something like:

```
{
  "vars": {
    "question": "question1",
    "document": "document1"
  }
}
```

**Comment by typpo** [comment]
Hi @maciejbrasewicz,

Interestingly, I'm unable to repro with the example you provided:

```
$ cat log_data.txt
Log data: {"vars":{"question":"question1","document":"document1"}}
Log data: {"vars":{"question":"question2","document":"document2"}}
```

Are you able to share more information about your environment? (promptfoo version, python version, OS, ...)

**Comment by maciejbrasewicz** [comment]
Promptfoo `0.24.1`
Python: `3.11.5`
Node: `v18.18.0`
OS: Windows 11

**Comment by typpo** [comment]
Thanks for the info, I think this is a Windows issue.  Potential fix in #179, though I do not have a Windows machine to confirm.

**Comment by typpo** [comment]
@maciejbrasewicz I've published a possible fix in version 0.24.2 - please give it a try and let me know if you're still not seeing the right args passed to the script.
